components/script/script_thread.rs

```diff
@@ -1596,12 +1634,12 @@ impl ScriptThread {
                 None
             });
 
-            // https://html.spec.whatwg.org/multipage/#event-loop-processing-model step 6
-            self.perform_a_microtask_checkpoint();
-
             if let Some(retval) = result {
                 return retval;
             }
+
+            // https://html.spec.whatwg.org/multipage/#event-loop-processing-model step 6
+            self.perform_a_microtask_checkpoint();
```

Comment:
note: strictly speaking not necessary, however it does make sense to not run microtasks if we're exiting the script-thread anyway(the `return retval` above).